scala-package/core/src/main/scala/org/apache/mxnet/NDArray.scala

```diff
@@ -105,6 +105,57 @@ object NDArray {
     })
   }
 
+  /**
+    * Used by NDArrayMacro for New Scala NDArray API.
+    * Invoke this function by passing in parameters.
+    * Parameters
+    * ----------
+    * @param kwargs Key-value arguments of input scalars
+    * @return The result NDArrays of result of computation.
+    */
+  private[mxnet] def genericNewAPINDArrayFunctionInvoke(
+    funcName: String, kwargs: Map[String, Any] = null) : NDArrayFuncReturn = {
+    val function = functions(funcName)
+    val ndArgs = ArrayBuffer.empty[NDArray]
+    val updatedKwargs: Map[String, String] =
+      Option(kwargs).getOrElse(Map.empty[String, String]).filter{ case (key, value) =>
+        !value.isInstanceOf[NDArray] && !value.isInstanceOf[NDArrayFuncReturn]
+      } .map { case (k, v) => k -> v.toString }
+
+    Option(kwargs).getOrElse(Map.empty[String, String]).filter{ case (key, value) =>
+          value.isInstanceOf[NDArray] || value.isInstanceOf[NDArrayFuncReturn]
+        } .filter{ case (key, value) => key != "out"}.foreach{
+      case (key, value) => value match {
+        case nd : NDArray =>
+          ndArgs.append(nd.asInstanceOf[NDArray])
+        case arrFunRet: NDArrayFuncReturn =>
+          arrFunRet.asInstanceOf[NDArrayFuncReturn].arr.foreach(ndArgs.append(_))
+      }
+    }
```

Comment:
looks like the `kwargs` is the only thing that is different from `genericNDArrayFunctionInvoke`. Can we still use `genericNDArrayFunctionInvoke` and make the change in `NDArrayMacro.scala` instead?

Reply:
It's working now, send the PR...

Reply:
If I didn't misunderstand, for `foo(arg1=ndarr1, arg2=ndarr2)`, macro generates a function call `genericNDArrayFunctionInvoke("foo", args=null, kwargs=Map("arg1"->ndarr1, "arg2"->ndarr2))`, then you assume  when iterating over `kwargs`, it is in the same order as you inserted into the Map, i.e., "arg1", "arg2" in this case - This is not correct:
```scala
scala> val map = scala.collection.mutable.Map[String, Any]()
map: scala.collection.mutable.Map[String,Any] = Map()
scala> map("hello") = 1
scala> map("apple")=2
scala> map
res2: scala.collection.mutable.Map[String,Any] = Map(apple -> 2, hello -> 1)
scala> map.toMap.foreach { case (k, v) => println(k) }
apple
hello
```
Can we simply create `ndArgs` in macros, and leave this function unchanged?

Reply:
I see your idea, two ways come in my mind. Change Macros design using Ndargs to store all NDArray element to match the order and store others into kv store. Second way I think we may need to find another C function to handle this method to ensure the correct order